internal/aws/awsutil/conn.go

```diff
@@ -267,24 +296,16 @@ func ProxyServerTransport(logger *zap.Logger, config *AWSSessionSettings) (*http
 	return transport, nil
 }
 
-func (c *Conn) newAWSSession(logger *zap.Logger, roleArn string, region string, profile string, sharedCredentialsFile []string) (*session.Session, error) {
+func (c *Conn) newAWSSession(logger *zap.Logger, cfg *AWSSessionSettings, region string) (*session.Session, error) {
```

Comment:
If we move the STS credential provider to the credential chain, then we can pass in the `*credential.Credentials` that we get from that chain instead of the `*AWSSessionSettings`.

Reply:
Role needs to be separated. As in if you give a role you do not use other creds.

Reply:
I think Jeffrey is saying add it to our custom order so it still takes precedence over the others (itll do it again as part of the fallback logic, but that might be ok?)

Reply:
I originally thought it could be an early return in the `getCredentialProviderChain` where it just returns a slice with the STS provider so we could have all our credential chain in one place, but looking at it again, that would break it. The STS provider requires a default session to get the role credentials in the first place so we'd end up in a circular dependency if the roleARN was present.